zebra-state/src/config.rs

```diff
@@ -136,124 +141,73 @@ impl Config {
-        // So we try setting the ideal limit, then the minimum limit.
-        if let Ok(actual_limit) =
-            Config::set_open_file_limit(Config::IDEAL_OPEN_FILE_LIMIT, hard_rlimit, old_limit)
-        {
-            return actual_limit;
-        }
-
-        // Try the hard limit or the minimum, whichever is greater
-        let min_limit =
-            if let Some(hard_limit) = hard_rlimit.map(TryInto::try_into).and_then(Result::ok) {
-                std::cmp::max(Config::MIN_OPEN_FILE_LIMIT, hard_limit)
-            } else {
-                Config::MIN_OPEN_FILE_LIMIT
-            };
-        if let Ok(actual_limit) = Config::set_open_file_limit(min_limit, hard_rlimit, old_limit) {
-            tracing::warn!(?actual_limit,
-                           ?hard_rlimit,
-                           ?old_limit,
-                           min_limit = ?Config::MIN_OPEN_FILE_LIMIT,
-                           ideal_limit = ?Config::IDEAL_OPEN_FILE_LIMIT,
-                           "the maximum number of open files is below Zebra's ideal limit. Hint: Increase the open file limit to {} before launching Zebra",
-                           Config::IDEAL_OPEN_FILE_LIMIT);
-            return actual_limit;
-        }
-
-        panic!("open file limit too low: unable to set the number of open files to {}, the minimum number of files required by Zebra. Current soft limit is {:?} and hard limit is {:?}. Hint: Increase the open file limit to {} before launching Zebra",
-               Config::MIN_OPEN_FILE_LIMIT,
-               old_limit,
-               hard_rlimit,
-               Config::IDEAL_OPEN_FILE_LIMIT);
-    }
-
-    /// Increase the soft open file limit for this process to `new_limit`,
-    /// and the hard open file limit to `hard_rlimit`.
-    ///
-    /// If `hard_rlimit` is `None`, also sets the hard limit to `new_limit`.
-    ///
-    /// If `old_limit` is already greater than or equal to `new_limit`,
-    /// returns `Ok(old_limit)`.
-    ///
-    /// Otherwise, tries to set the limit. Returns `Ok(new_limit)` if the
-    /// limit is set successfully.
-    #[cfg(unix)]
-    fn set_open_file_limit(
-        new_limit: usize,
-        hard_rlimit: Option<rlimit::Rlim>,
-        old_limit: Option<usize>,
-    ) -> Result<usize, ()> {
-        use rlimit::{setrlimit, Resource};
+    fn increase_open_file_limit() -> u64 {
+        // `increase_nofile_limit` doesn't do anything on Windows in rlimit 0.7.0.
+        //
+        // On Windows, the default limit is:
+        // - 512 high-level stream I/O files (via the C standard functions), and
+        // - 8192 low-level I/O files (via the Unix C functions).
+        // https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/setmaxstdio?view=msvc-160#remarks
+        //
+        // If we need more high-level I/O files on Windows,
+        // use `setmaxstdio` and `getmaxstdio` from the `rlimit` crate:
+        // https://docs.rs/rlimit/latest/rlimit/#windows
```

Comment:
Question: I'm a bit confused about this paragraph. Does it describe something we should do in the future if we need to increase the limit? Or does it describe what is currently done somewhere else to increase the limit on Windows?

If it's the former, I might suggest a slight tweak to the phrasing:

```suggestion
        // If we ever need more high-level I/O files on Windows,
        // we will have to use `setmaxstdio` and `getmaxstdio` from the `rlimit` crate:
        // https://docs.rs/rlimit/latest/rlimit/#windows
```

Reply:
It probably won't be needed, but I should explain why.

Fixed in PR #3590.